qiime/plugin/testing.py

```diff
@@ -84,3 +90,25 @@ def assertSemanticTypeRegisteredToFormat(self, semantic_type, exp_format):
             obs_format, exp_format,
             "Expected semantic type %r to be registered to format %r, not %r."
             % (semantic_type, exp_format, obs_format))
+
+    def transform_format(self, source_format, target, filenames):
```

Comment:
Since we are supporting both `FileFormat` and `DirectoryFormat`, I think we need to support providing one `filename` or a list of `filenames`. How about we make those `kwargs`, and put an exclusivity guard in (one or the other). Motivation for this is described in a few of the following comments.